### PR TITLE
Feature/move to black linting

### PR DESCRIPTION
**Pull request by sampsapenna**
### Description

A tool called black is used as an automatic linter for Python projects. This PR moves the `swift-browser-ui` to use the coding style of Black.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (this needs a follow up PR)

### Changes Made

* Move to black coding style
* Change tox flake8 to use black coding style for style checks

### Testing

- [ ] Unit Tests
- [ ] Integration Tests
- [x] Tests do not apply
- [ ] Needs testing (start an issue or do a follow up PR about it)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -5,56 +5,60 @@
...
     install_requires=[
-        'aiohttp',
-        'python-swiftclient',
-        'cryptography',
-        'keystoneauth1',
-        'click',
-        'gunicorn',
-        'uvloop',
-        'certifi'
+        "aiohttp",
```
Comment (blankdots) [inline setup.py:16]: can we pin versions on these as well or we can follow metadata-submitter and load `requirements.txt` file https://github.com/CSCfi/metadata-submitter/blob/master/setup.py

File: setup.py [hunk 2]
```diff
@@ -5,56 +5,60 @@
 setuptools.setup(
     name=__name__,
     version=__version__,
-    description='Object browser for OS_swift API',
+    description="Object browser for OS_swift API",
     author=__author__,
     author_email="sampsa.penna@csc.fi",
```
Comment (blankdots) [inline setup.py:10]: would be good to have our support email here or leave out.

File: swift_browser_ui/_convenience.py [hunk 3]
```diff
@@ -48,12 +48,12 @@ def test_swift_endpoint(endpoint: str) -> None:
             reason="Cannot get Swift endpoint connection."
         )
     else:
-        logging.info('Swift endpoint accessible ')
+        logging.info("Swift endpoint accessible ")
```
Comment (blankdots) [inline swift_browser_ui/_convenience.py:51]:
```suggestion
        logging.info("Swift endpoint accessible.")
```

File: swift_browser_ui/api.py [hunk 4]
```diff
@@ -15,46 +15,39 @@
...
-        tenant = f"Container: {item['container']}" \
-            if item['container'] \
+        tenant = (
+            f"Container: {item['container']}"
+            if item["container"]
             else "No tenant specified, working with container"
-        request.app['Log'].info(f"{tenant} "
-                                "list unpacked successfully.")
+        )
+        request.app["Log"].info(f"{tenant} " "list unpacked successfully.")
```
Comment (blankdots) [inline swift_browser_ui/api.py:44]:
```suggestion
        request.app["Log"].info(f"{tenant} list unpacked successfully.")
```

File: swift_browser_ui/api.py [hunk 5]
```diff
@@ -15,46 +15,39 @@
...
-async def get_os_user(
-        request: aiohttp.web.Request
-) -> aiohttp.web.Response:
+async def get_os_user(request: aiohttp.web.Request) -> aiohttp.web.Response:
     """Fetch the session owning OS user."""
     session = api_check(request)
-    request.app['Log'].info(
-        'API call for username from {0}, sess: {1} :: {2}'.format(
+    request.app["Log"].info(
+        "API call for username from {0}, sess: {1} :: {2}".format(
```
Comment (blankdots) [inline swift_browser_ui/api.py:22]: f strings are also an option, considering that in some places we use f string and in some `.format` we need to pick one. I think f strings provider a cleaner syntax but also ok with `.format`. The same in other places.

File: swift_browser_ui/api.py [hunk 6]
```diff
@@ -138,45 +119,44 @@ async def swift_list_objects(
...
+                i["content_type"] = "binary/octet-stream"
             else:
-                i['content_type'] = i['content_type'].replace('\u0000', '')
+                i["content_type"] = i["content_type"].replace("\u0000", "")
 
         return aiohttp.web.json_response(obj)
     except SwiftError:
-        request.app['Log'].error("SwiftError occured "
-                                 "return empty container list.")
+        request.app["Log"].error("SwiftError occured " "return empty container list.")
```
Comment (blankdots) [inline swift_browser_ui/api.py:150]:
```suggestion
        request.app["Log"].error("SwiftError occurred return empty container list.")
```

File: swift_browser_ui/api.py [hunk 7]
```diff
@@ -138,45 +119,44 @@ async def swift_list_objects(
...
         return aiohttp.web.json_response(obj)
     except KeyError:
         # listing is missing; possible broken swift auth
-        request.app['Log'].error("listing is missing; "
-                                 "possible broken swift auth.")
+        request.app["Log"].error("listing is missing; " "possible broken swift auth.")
```
Comment (blankdots) [inline swift_browser_ui/api.py:154]:
```suggestion
        request.app["Log"].error("listing is missing; possible broken swift auth.")
```

File: swift_browser_ui/api.py [hunk 8]
```diff
@@ -192,118 +172,106 @@ async def swift_list_shared_objects(
...
             else:
-                i['content_type'] = i['content_type'].replace('\u0000', '')
+                i["content_type"] = i["content_type"].replace("\u0000", "")
 
         return aiohttp.web.json_response(obj)
 
     except SwiftError:
-        request.app['Log'].error("SwiftError occured "
-                                 "return empty container list.")
+        request.app["Log"].error("SwiftError occured " "return empty container list.")
```
Comment (blankdots) [inline swift_browser_ui/api.py:202]:
```suggestion
        request.app["Log"].error("SwiftError occurred return empty container list.")
```

File: swift_browser_ui/api.py [hunk 9]
```diff
@@ -192,118 +172,106 @@ async def swift_list_shared_objects(
...
         return aiohttp.web.json_response(obj)
     except ClientException as e:
-        request.app['Log'].error(e.msg)
+        request.app["Log"].error(e.msg)
         return aiohttp.web.json_response(obj)
     except KeyError:
         # listing is missing; possible broken swift auth
-        request.app['Log'].error("listing is missing; "
-                                 "possible broken swift auth.")
+        request.app["Log"].error("listing is missing; " "possible broken swift auth.")
```
Comment (blankdots) [inline swift_browser_ui/api.py:209]:
```suggestion
        request.app["Log"].error("listing is missing; possible broken swift auth.")
```

File: swift_browser_ui/api.py [hunk 10]
```diff
@@ -192,118 +172,106 @@ async def swift_list_shared_objects(
...
-    serv = request.app['Sessions'][session]['ST_conn']
-    sess = request.app['Sessions'][session]['OS_sess']
+    serv = request.app["Sessions"][session]["ST_conn"]
+    sess = request.app["Sessions"][session]["OS_sess"]
 
     temp_url_key = await get_tempurl_key(serv)
-    request.app['Log'].debug(
-        "Using %s as temporary URL key", temp_url_key
-    )
+    request.app["Log"].debug("Using %s as temporary URL key", temp_url_key)
```
Comment (blankdots) [inline swift_browser_ui/api.py:228]: this is neither f string nor format ... need to standardise , check other places as well.

File: swift_browser_ui/api.py [hunk 11]
```diff
@@ -76,58 +69,46 @@ async def swift_list_buckets(
...
         if not cont:
             # return empty object
-            request.app['Log'].debug("Empty container list.")
+            request.app["Log"].debug("Empty container list.")
             raise aiohttp.web.HTTPNotFound()
         return aiohttp.web.json_response(cont)
     except SwiftError:
-        request.app['Log'].error("SwiftError occured "
-                                 "return empty container list.")
+        request.app["Log"].error("SwiftError occured " "return empty container list.")
```
Comment (blankdots) [inline swift_browser_ui/api.py:81]:
```suggestion
        request.app["Log"].error("SwiftError occurred return empty container list.")
```

File: swift_browser_ui/api.py [hunk 12]
```diff
@@ -76,58 +69,46 @@ async def swift_list_buckets(
...
         raise aiohttp.web.HTTPNotFound()
     except KeyError:
         # listing is missing; possible broken swift auth
-        request.app['Log'].error("listing is missing; "
-                                 "possible broken swift auth.")
+        request.app["Log"].error("listing is missing; " "possible broken swift auth.")
```
Comment (blankdots) [inline swift_browser_ui/api.py:85]:
```suggestion
        request.app["Log"].error("listing is missing; possible broken swift auth.")
```

File: swift_browser_ui/api.py [hunk 13]
```diff
@@ -613,43 +548,42 @@ async def get_project_metadata(
...
         return aiohttp.web.json_response(ret)
     except ClientException as e:
-        request.app['Log'].error(e.msg)
+        request.app["Log"].error(e.msg)
         return aiohttp.web.json_response(ret)
     except KeyError:
-        request.app['Log'].error("items is missing; possible allas "
-                                 "is not authorised for project.")
+        request.app["Log"].error(
+            "items is missing; possible allas " "is not authorised for project."
```
Comment (blankdots) [inline swift_browser_ui/api.py:577]:
```suggestion
            "items is missing; possible swift storage is not authorised for project."
```
let us keep the `Allas` name out of this app

File: swift_browser_ui/api.py [hunk 14]
```diff
@@ -837,59 +757,50 @@ async def remove_container_acl(
...
     )
-    serv = request.app['Sessions'][session]['ST_conn']
+    serv = request.app["Sessions"][session]["ST_conn"]
 
     container = request.match_info["container"]
     projects = request.query["projects"].split(",")
-    request.app['Log'].debug(f"Requested container {container} "
-                             f"and projects {projects}.")
+    request.app["Log"].debug(
+        f"Requested container {container} " f"and projects {projects}."
```
Comment (blankdots) [inline swift_browser_ui/api.py:783]:
```suggestion
        f"Requested container {container} and projects {projects}."
```

File: swift_browser_ui/server.py [hunk 15]
```diff
@@ -238,40 +237,37 @@ def run_server_secure(
...
-        app: typing.Coroutine[typing.Any, typing.Any, aiohttp.web.Application]
+    app: typing.Coroutine[typing.Any, typing.Any, aiohttp.web.Application]
 ) -> None:
     """Run the server without https enabled."""
     aiohttp.web.run_app(
         app,
-        access_log=aiohttp.web.logging.getLogger('aiohttp.access'),
-        port=(setd['port'])  # type: ignore
+        access_log=aiohttp.web.logging.getLogger("aiohttp.access"),
+        port=(setd["port"]),  # type: ignore
```
Comment (blankdots) [inline swift_browser_ui/server.py:266]: we could make this `int` instead of ignoring type

File: tests/unit/coveralls.py [hunk 16]
```diff
@@ -8,9 +8,9 @@
 # Solution provided by
 # https://stackoverflow.com/questions/32757765/conditional-commands-in-tox-tox-travis-ci-and-coveralls
 
-if __name__ == '__main__':
-    if 'COVERALLS_REPO_TOKEN' in os.environ:
-        rc = call('coveralls')
+if __name__ == "__main__":
+    if "COVERALLS_REPO_TOKEN" in os.environ:
+        rc = call("coveralls")
         sys.stdout.write("Coveralls report from TRAVIS CI.\n")
```
Comment (blankdots) [inline tests/unit/coveralls.py:14]:
```suggestion
        sys.stdout.write("Coveralls report from CI.\n")
```
We don't use Travis any more

**Review by blankdots — CHANGES_REQUESTED** [review]
It is a request for change, mostly to also address the fact that we don't follow a uniform styling for logs and other things.
Some misses with `" "` that black formatting did not handle.
Would be good to pick one of and stick with it always:
- f strings
-` %` strings
- `.format` strings

Other misses

**Comment by blankdots** [comment]
also `black . -l 90 --check` still shows some files that would nee reformatting - and would be good if we can check that via CI so that we know we follow a certain style guide.

File: swift_browser_ui/server.py (hunk 15 above)
Comment (sampsapenna) [reply swift_browser_ui/server.py:266]: The port is an integer in the dictionary – the settings dictionary could be moved over to a TypedDict to remove this issue, which would be a better solution in my opinion. This, however, would require dropping support for Python < 3.8 (TypedDict comes built-in from 3.8 onwards).
Forcing the integer with `int()` only generates a different mypy error.

File: swift_browser_ui/api.py (hunk 5 above)
Comment (sampsapenna) [reply swift_browser_ui/api.py:22]: moving over to f-strings with next push to this branch – unifying string style is a good thing

File: swift_browser_ui/server.py (hunk 15 above)
Comment (blankdots) [reply swift_browser_ui/server.py:266]: i am fine with dropping support for <3.8, but would postpone the Typedict for a different issue

File: setup.py (hunk 2 above)
Comment (sampsapenna) [reply setup.py:10]: True, removing the support email for now – it can be added back later

Comment (sampsapenna) [reply setup.py:10]: As I think that the support for software itself should be gotten on GitHub, and other (mainly usage) support should be given in the actual service

**Comment by sampsapenna** [comment]
Black style check has been added to the push-workflow on github, and `tox.ini`

File: swift_browser_ui/_convenience.py [hunk 17]
```diff
@@ -318,29 +270,28 @@ def get_availability_from_token(
...
         headers=hdr,
     )
     with urllib.request.urlopen(drq, timeout=20) as domains:  # nosec
-        output_domains = json.loads(domains.read().decode('utf-8'))
+        output_domains = json.loads(domains.read().decode("utf-8"))
 
-    logging.info("%s\n%s",
-                 str(output_projects),
-                 str(output_domains))
+    logging.info("%s\n%s", str(output_projects), str(output_domains))
```
Comment (blankdots) [inline swift_browser_ui/_convenience.py:287]:
```suggestion
    logging.info(f"{str(output_projects)}\n{str(output_domains)}")
```

File: swift_browser_ui/login.py [hunk 18]
```diff
@@ -27,333 +27,272 @@
...
     session = ""
-    log = request.app['Log']
-    if not setd['set_session_devmode']:
+    log = request.app["Log"]
+    if not setd["set_session_devmode"]:
         try:
             session = decrypt_cookie(request)["id"]
-            log.info("Killing session for %s :: %s",
-                     session, time.ctime())
+            log.info("Killing session for %s :: %s", session, time.ctime())
```
Comment (blankdots) [inline swift_browser_ui/login.py:289]:
```suggestion
            log.info(f"Killing session for {session} :: {time.ctime()}")
```

File: swift_browser_ui/login.py [hunk 19]
```diff
@@ -27,333 +27,272 @@
...
         except aiohttp.web.HTTPUnauthorized:
-            log.info(
-                "Trying to log out an invalidated session: {0}".format(
-                    session
-                )
-            )
-    response = aiohttp.web.Response(
-        status=303
-    )
+            log.info("Trying to log out an invalidated session: {0}".format(session))
```
Comment (blankdots) [inline swift_browser_ui/login.py:295]:
```suggestion
            log.info(f"Trying to log out an invalidated session: {session}")
```

File: tests/unit/test_login.py [hunk 20]
```diff
@@ -25,40 +25,44 @@ class LoginTestClass(asynctest.TestCase):
...
-                "?origin={origin}".format(
-                    origin="https://localhost/login/websso"
-                )
-            ))
+            self.assertEqual(
+                resp.headers["Location"],
+                (
+                    "https://example.os.com:5001/v3"
+                    + _path
+                    + "?origin={origin}".format(origin="https://localhost/login/websso")
```
Comment (blankdots) [inline tests/unit/test_login.py:51]:
```suggestion
                    + "?origin=https://localhost/login/websso"
```
we don't really need a formatting here

**Review by blankdots — APPROVED** [review]

File: swift_browser_ui/_convenience.py (hunk 17 above)
Comment (sampsapenna) [reply swift_browser_ui/_convenience.py]: bundled these suggestions to a single commit

**Pull request merged**